### ISSUE TITLE
mef_eline isn't allocating/releasing `available_tags` for UNIs

### ISSUE THREAD

**Issue by viniarck** [kytos-ng/mef_eline#324]
When a EVPL is created, only NNIs are allocating vlans from `available_tags`, `UNIs` vlans are entirely relying on `_is_duplicated_evc(evc)` to try to detect as conflict as mentioned on issue #323, which also works, but this is an inconsistency implementation detail that makes things hard especially in the future when vlan range will be introduced, if we were to coherently also be allocating and releasing `available_vlans` for UNIs when an EVC is being created/deleted/updated, then to check if a VLAN is still available to be created is just a matter of checking if the tag or range of tags are available, which is simpler than iterating over all circuits and still compare (which is also OK, even though it also has a worse time complexity but as vlan ranges are introduced it's becoming evident for vlan ranges the current implementation will make things harder than it should be, so it's probably worth it allocating/releasing the VLANs for UNIs when tags are in place to simplify code maintenance)

**Comment by italovalcy** [comment]
@viniarck we are planing to use this feature in production, and the potential risks of having overlapping usage of a tag for NNI/UNI is quite important. I believe we should increase the priority of this issue and even change it's label to a bug instead of enhancement (since it creates the possibility of duplication, I would consider as a bug)

**Comment by italovalcy** [comment]
As a workaround, we could do the following on the kytos console:

```
kytos $> from kytos.core.interface import TAG, TAGType
kytos $> controller.switches["00:00:00:00:00:00:00:01"].interfaces[2].use_tag(TAG(TAGType.VLAN, 1899))
Out[21]: True
```

**Comment by italovalcy** [comment]
@viniarck @Alopalao we are starting to have some problems related to this issue in a deploy we have and we might need to think about a workaround solution for this problem (backporting to previous releases). The problems we are seeing is due to NNI being used as UNIs to some EVCs, and the pool of VLANs are reaching the range of user-vlans, which are creating conflicting flows for Service VLANs and for Customer VLANs.

**Comment by viniarck** [comment]
@italovalcy, right. @Alopalao's PR https://github.com/kytos-ng/mef_eline/pull/371, is also taking an opportunity to fix another update issue, and it's also leveraging his other PR on kytos https://github.com/kytos-ng/kytos/pull/409 that's fixing vlan pool and using the new methods (that are already being prepared to also work with a range of tags) when allocating and releasing vlans.

If only allocating/deallocating the vlans solve the problem then we could backport only this part to `2023.1` (and potentially to `2022.3` if you can't afford to upgrade to `2023.1` soon), and then provide a migration script to update DB collection `interface_details` `available_vlans` accordingly. Does it solve the problem or you need more than this? @Alopalao if you could make an assessment here or if you have other ideas, thanks. 

Side note: backporting all of the data model changes would be very difficult and it's a significant effort in terms of development and exploratory tests (since it impacts horizontally `mef_eline` and `topology` too). We'll need to find a balance in terms of effort and minimizing risks while providing a solution for this problem (and if we need more than the previous paragraph then also assess if we either cut the release early and or also check the viability or an anticipated prod deployment).

**Comment by Alopalao** [comment]
Talked about this with Italo I think a temporal solution would be better for this case. `kytos`, `mef_eline` and `topology` would need to have these changes so UNI can get their TAGs allocated. This way we will reduce the process of making sure the changes with data model and changes with lists of ranges of TAGs are working correctly. These changes are doomed to be deprecated later though.
